src/queryMethod.ts

```diff
@@ -4,20 +4,26 @@ import { logger } from './logSettings';
 import type { Func, QueryMethodMap } from './types';
 
 /**
- * Define Options for the Class
- * @param options Options
- * @example Example:
- * ```
- *  @modelOptions({ schemaOptions: { timestamps: true } })
- *  class Name {}
+ * Adds a query method to schema.
+ *
+ * @param func Query function
+ * @example
+ * ```ts
+ * function findByTitle(this: ReturnModelType<typeof Event>, title: string) {
+ *  return this.find({ title });
+ * }
  *
- *  // Note: The default Class "TimeStamps" can be used for type information and options already set
+ * @queryMethod(findByTitle)
+ * class Event {
+ *  @prop()
+ *  public title: string;
+ * }
  * ```
  */
 export function queryMethod(func: Func) {
   return (target: any) => {
     logger.info('Adding query method "%s" to %s', func.name, getName(target));
-    const queryMethods: QueryMethodMap = new Map(Reflect.getMetadata(DecoratorKeys.QueryMethod, target.constructor) ?? []);
+    const queryMethods: QueryMethodMap = new Map(Reflect.getMetadata(DecoratorKeys.QueryMethod, target) ?? []);
```

Comment:
why remove the `.constructor` here?

Reply:
Because it's directly being applied to `target` on line 28:
```ts
Reflect.defineMetadata(DecoratorKeys.QueryMethod, queryMethods, target);
```

Also, it's being retrieved from the class itself in `internal/schema.ts` on line 118.

Reply:
thanks didnt notice this was the only place where i used `constructor`